lib/shopify_api/event_pipe/background_job_behaviour.ex

```diff
@@ -0,0 +1,29 @@
+defmodule ShopifyAPI.EventPipe.BackgroundJobBehaviour do
+  @moduledoc """
+  Defines what an implementation for a background job handler can be.
+
+  By default Exq will be used and needs to be configured.
+
+  You can set the implmentation with `Applicaiton.set_env(:shopify_api, :background_job_implementation, MyImplementation)`
+
+  The implementation is respnsible for calling `worker.perform(event)` for every event sent to it.
+  """
+
+  alias ShopifyAPI.AuthToken
+
+  @doc """
+  Subscribes to the background job implementation with the given AuthToken
+  This allows for multiple sites to be run each with their own queue.
+  """
+  @callback subscribe(token :: %AuthToken{}) :: no_return()
+
+  @doc """
+  Sends an event to the Queue.
+  """
+  @callback enqueue(
+              queue_name :: String.t(),
+              worker :: module(),
+              events :: list(ShopifyAPI.EventPipe.Event.t()),
+              opts :: any()
+            ) :: no_return()
```

Comment:
should subscribe be in here too?

Reply:
Subscribe seems to be dependent on the background job implementation, so I think it is required in here.